workers/unity/Packages/io.improbable.gdk.debug/.codegen/Source/Generators/DebugExtensions/ComponentVisualElementGenerator.cs

```diff
@@ -22,28 +29,184 @@ public static CodeWriter Generate(UnityComponentDetails details)
                     {
                         type.Line($"public override ComponentType ComponentType {{ get; }} = ComponentType.ReadOnly<{details.Name}.Component>();");
 
+                        type.TextList(details.FieldDetails.Select(ToFieldDeclaration));
+
                         GenerateConstructor(type, details);
                         GenerateUpdateMethod(type, details);
                     });
                 });
             });
         }
 
+        private static string ToFieldDeclaration(UnityFieldDetails fieldDetails)
+        {
+            switch (fieldDetails.FieldType)
+            {
+                case SingularFieldType singularFieldType:
+                    var uiType = GetUiFieldType(singularFieldType.ContainedType);
+
+                    if (uiType == "")
+                    {
+                        // TODO: Eliminate this case.
+                        return "";
+                    }
+
+                    return $"private readonly {uiType} {fieldDetails.CamelCaseName}Field;";
+                default:
+                    // TODO: Lists, maps, and options
+                    return "";
+            }
+        }
+
         private static void GenerateConstructor(TypeBlock typeBlock, UnityComponentDetails details)
         {
             typeBlock.Method($"public {details.Name}Renderer() : base()", mb =>
             {
                 mb.Line($"ComponentFoldout.text = \"{details.Name}\";");
                 mb.Line($"AuthoritativeToggle.SetEnabled(false);");
+
+                foreach (var field in details.FieldDetails)
+                {
+                    mb.TextList(ToFieldInitialisation(field));
+                }
             });
         }
 
+        private static IEnumerable<string> ToFieldInitialisation(UnityFieldDetails fieldDetails)
+        {
+            switch (fieldDetails.FieldType)
+            {
+                case SingularFieldType singularFieldType:
+
+                    var uiType = GetUiFieldType(singularFieldType.ContainedType);
+
+                    if (uiType == "")
+                    {
+                        // TODO: Eliminate this case.
+                        yield break;
+                    }
+
+                    var humanReadableName = Formatting.SnakeCaseToHumanReadable(fieldDetails.Name);
+                    yield return $"{fieldDetails.CamelCaseName}Field = new {uiType}(\"{humanReadableName}\");";
```

Comment:
Extract the string into a variable: `{fieldDetails.CamelCaseName}Field`

Reply:
I've refactored this a little bit in the user-types branch, will apply this suggestion there 👍